modules/apps/permission-template/permission-template-service/src/main/java/com/liferay/permission/template/service/impl/PermissionTemplateLocalServiceImpl.java

```diff
@@ -23,13 +23,52 @@
 public class PermissionTemplateLocalServiceImpl
 	extends PermissionTemplateLocalServiceBaseImpl {
 
-	public PermissionTemplate fetchPermissionTemplateSetting(
+	public PermissionTemplate addPermissionTemplate(
+		long companyId, long groupId, String className, long classPK,
+		boolean permissionTemplateEnabled) {
+
+		long permissionTemplateId = counterLocalService.increment();
+
+		PermissionTemplate permissionTemplate =
+			permissionTemplatePersistence.create(permissionTemplateId);
+
+		permissionTemplate.setGroupId(groupId);
+		permissionTemplate.setCompanyId(companyId);
+		permissionTemplate.setClassNameId(
+			_classNameLocalService.getClassNameId(className));
+		permissionTemplate.setClassPK(classPK);
+		permissionTemplate.setPermissionTemplateEnabled(
+			permissionTemplateEnabled);
+
+		return permissionTemplatePersistence.update(permissionTemplate);
+	}
+
+	public PermissionTemplate fetchPermissionTemplate(
 		String className, long classPK) {
 
 		return permissionTemplatePersistence.fetchByC_C(
 			_classNameLocalService.getClassNameId(className), classPK);
 	}
 
+	public PermissionTemplate updatePermissionTemplate(
+		long companyId, long groupId, String className, long classPK,
+		boolean permissionTemplateEnabled) {
+
+		PermissionTemplate permissionTemplate = fetchPermissionTemplate(
```

Comment:
Only one `PermissionTemplate` can exist with the key `(className, classPK)`? Or is it possible to have different ones depending on the `companyId` or `groupId`?